lib/igb/igb.c

```diff
@@ -1093,6 +1094,12 @@ igb_clean(device_t *dev, struct igb_packet **cleaned_packets)
 		if (txr->tx_avail >= IGB_QUEUE_THRESHOLD)	  
 			txr->queue_status &= ~IGB_QUEUE_DEPLETED;
 	}	
+
+unlock:
```

Comment:
I don't see anything using this label in the function, and the conditional wrapping the sem_post() seems to not be serving any purpose either. I understand the desire to check return values, but it seems pointless to me to add a check that doesn't DO anything different based on the value. I guess it makes it obvious that it should be an early return if other code gets added on?

These concerns shouldn't affect the functionality of the code so I don't want to block it if it needs to go in and others think it's fine, but they did leave me scratching my head a bit as I reviewed the code.